projects/igniteui-angular/src/lib/directives/autocomplete/autocomplete.directive.ts

```diff
@@ -0,0 +1,385 @@
+    positionStrategy?: IPositionStrategy;
+    /** Scroll strategy to use with this settings */
+    scrollStrategy?: IScrollStrategy;
+    /** Set the outlet container to attach the overlay to */
+    outlet?: IgxOverlayOutletDirective | ElementRef;
+}
+
+/**
+ * **Ignite UI for Angular Autocomplete** -
+ * [Documentation](https://www.infragistics.com/products/ignite-ui-angular/angular/components/autocomplete.html)
+ *
+ * The igxAutocomplete directive provides a way to enhance a text input
+ * by showing a drop down of suggested options, provided by the developer.
+ *
+ * Example:
+ * ```html
+ * <input type="text" [igxAutocomplete]="townsPanel" />
+ * <igx-drop-down #townsPanel>
+ *     <igx-drop-down-item *ngFor="let town of towns | startsWith:townSelected" [value]="town">
+ *         {{town}}
+ *     </igx-drop-down-item>
+ * </igx-drop-down>
+ * ```
+ */
+@Directive({
+    selector: '[igxAutocomplete]'
+})
+export class IgxAutocompleteDirective extends IgxDropDownItemNavigationDirective implements OnDestroy {
+
+    constructor(@Self() @Optional() @Inject(NgModel) protected ngModel: NgModel,
+                @Self() @Optional() @Inject(FormControlName) protected formControl: FormControlName,
+                @Optional() protected group: IgxInputGroupComponent,
+                protected elementRef: ElementRef,
+                protected cdr: ChangeDetectorRef) {
+        super(null);
+    }
+
+    private _disabled = false;
+    private settings: OverlaySettings = {
+        modal: false,
+        scrollStrategy: new AbsoluteScrollStrategy(),
+        positionStrategy: new AutoPositionStrategy({ target: this.parentElement })
+    };
+
+    protected id: string;
+    protected dropDownOpened$ = new Subject<boolean>();
+    protected get model() {
+        return this.ngModel || this.formControl;
+    }
+
+    /**
+     * @hidden
+     * @internal
+     */
+    get nativeElement(): HTMLInputElement {
+        return this.elementRef.nativeElement;
+    }
+
+    /**
+     * @hidden
+     * @internal
+     */
+    get parentElement(): HTMLElement {
+        return this.group ? this.group.element.nativeElement : this.nativeElement;
+    }
+
+    /**
+     * @hidden
+     * @internal
+     */
+    target;
+
+    /**
+     * @hidden
+     * @internal
+     */
+    @Input('igxAutocomplete')
+    public dropdown: IgxDropDownComponent;
+
+    /**
+     * Enables/disables autocomplete component
+     *
+     * ```typescript
+     * // get
+     * let disabled = this.autocomplete.disabled;
+     * ```
+     * ```html
+     * <!--set-->
+     * <input type="text" [igxAutocomplete]="townsPanel" [igxAutocompleteDisabled]="disabled"/>
+     * ```
+     * ```typescript
+     * // set
+     * public disabled = true;
+     * ```
+     */
+    @Input('igxAutocompleteDisabled')
+    get disabled() {
+        return this._disabled;
+    }
+    set disabled(value) {
+        this._disabled = value;
+        this.close();
```

Comment:
No need for the close and get/set. It's API-only, close can be called if needed when setting this.